BEPs/BEP-335.md

```diff
@@ -0,0 +1,147 @@
+However, it is worth noting that even though the exit SP may cooperate, transferring a significant amount of data to all successors can be challenging to predict or manage effectively.
+The graph below illustrates the necessary operations that each component must complete when the SP needs to exit as the primary SP from a GVG.
+
+![](./assets/BEP-335/4-3-new-sp-graceful-exit-procedure.png)
+
+1. The SP initiates a StorageProviderExit transaction to Greenfield, declares itself Graceful_Exiting status, After this, it will no longer serve any new bucket requests, and existing buckets will not receive new objects. Additionally, it will not serve as a secondary SP for any new objects. However, user query requests will still be accepted.
+  Note: Once the graceful exit is started, it can not be canceled.
+2. Other interested SPs who wish to take over VGF and GVG can send a ReserveSwapIn transaction to Greenfield. This reservation will expire in 7 days. During this period, the prospective successor SP will be allowed to acquire data from secondary SPs( If the exiting SP served as a secondary SP, the prospective successor SP will acquire data from the primary SP)
+3. The prospective successor SP will continually request data recovery from these secondary SPs. The secondary SPs will verify the validity of the successor SP against Greenfield. Additionally, the successor SP will verify the received data against Greenfield to ensure its integrity.
+4. Once the successor SP successfully acquires the data and verifies its integrity, it will send a CompleteSwapIn transaction to acknowledge the success. This will officially replace the exiting SP in the VGF/GVG.
+5. Once the exiting SP no longer has any GVG bind, it can send out a CompleteStorageProviderExit transaction to signify its completion of the exit process.
+
+### 4.3.1 Data heal process
+
+As described in the current SP graceful exit procedure, one of the significant challenges is the SP conflict issue, let’s discuss it in more detail. Let's delve into this matter in more detail using an example.
+
+Consider a GVG with SPs {SP1, SP2, SP3, SP4, SP5, SP6, SP7}, where SP1 is the primary SP.
+If SP3 needs to become the successor, it must first be swapped out from the GVG by a successor, let's say SP8. The data transfer from SP3 to SP8 will take place before the transfer from SP1 to SP8. However, the new implementation allows the successor SP to take the primary SP role, even if it is currently a secondary SP in the GVG.  This means that a GVG can temporarily break the redundancy requirement, and data redundancy can be achieved through a healing process.
```

Comment:
"The data transfer from SP3 to SP8 will take place before the transfer from SP1 to SP8", shouldn't it be "The data transfer from SP3 to SP8 will take place before the transfer from SP1 to SP3"?

Reply:
Yes, fixed.